src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ComponentResourceManager.cs

```diff
@@ -149,10 +149,10 @@ public virtual void ApplyResources(object value, string objectName, CultureInfo
                     }
                 }
 
-                // Character after objectName.Length should be a ".", or else we should continue.
+                // Character after objectName.Length should be a "." or a '-', or else we should continue.
```

Comment:
Yes, when using - works fine. I did - because as we talked underscore is valid within identifiers and underscore is not. Just to use a not valid char within identifiers.